### PR TITLE
e.io scheduler

### PR DESCRIPTION
**Pull request by pnedelko**

### CODE REVIEW

File: services/scheduler/src/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    constructor(config, flowsDao, schedulePublisher) {
+        this._logger = defaultLogger;
+        this._config = config;
+        this._flowsDao = flowsDao;
+        this._schedulePublisher = schedulePublisher;
+    }
+
+    async _scheduleFlows() {
+        const flows = await this._flowsDao.findForScheduling({
+            limit: parseInt(this._config.get('POLLING_TASK_COUNT'))
```
Comment (pnedelko) [inline services/scheduler/src/index.js]: this limit could be a part of an implementation, not a framework. How do you think, @anton-kotenko ?

File: src/scheduler/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "scripts": {
+    "pretest": "eslint ./src",
+    "test": "mocha spec --recursive"
+  },
+  "dependencies": {
+    "bunyan": "1.8.12"
+  },
+  "devDependencies": {
+    "chai": "^4.2.0",
```
Comment (anton-kotenko) [inline src/scheduler/package.json]: plz don't use ^ versions. This is possible source of problems

Comment (pnedelko) [reply src/scheduler/package.json]: @anton-kotenko fixed

File: src/scheduler/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Default implementation is to be implemented yet
+There are 3 objects that a `Scheduler` instance requires: `config`, `flowsDao` and `schedulePublisher`. These objects have to be provided to the constructor of `Scheduler`.
+`flowsDao` is a service, where `Scheduler` finds flows for execution.
+`schedulePublisher` is a service which is supposed to publish a command for scheduling a certain flow.
+
+### Elastic.io implementation
+As an example, `elastic.io` implementation of the mentioned dependencies:
+
+```
+const flowScheduler = new FlowScheduler({ amqpConnection: init.getAMQPConnection() });
```
Comment (anton-kotenko) [inline src/scheduler/README.md:13]: I don't sure that that is good example. It is too deeply integrated with elastic.io specific.
I think the best possible example is code => use old code that works with k8s CRD's for dao and src/backendCommonsLib/src/QueueCreator.js as schedulePublisher

File: services/scheduler/Dockerfile [hunk 4]
```diff
@@ -2,10 +2,9 @@ FROM node:8.12-alpine AS base
 WORKDIR /usr/src/scheduler
 
 COPY package.json yarn.lock ./
-COPY src/backendCommonsLib src/backendCommonsLib
-COPY services/scheduler/package.json services/scheduler/index.js ./services/scheduler/
-COPY services/scheduler/src ./services/scheduler/src
-COPY services/scheduler/config/default.json services/scheduler/config/default.json
+COPY services/scheduler/package.json services/scheduler/app.js ./services/scheduler/
```
Comment (anton-kotenko) [inline services/scheduler/Dockerfile]: need to attach src/scheduler here. Or build would fail

File: services/scheduler/src/index.js (hunk 1 above)
Comment (anton-kotenko) [reply services/scheduler/src/index.js]: discussed

**Review by anton-kotenko — APPROVED** [review]

**Comment by pnedelko** [comment]
Closing if favour of https://github.com/openintegrationhub/openintegrationhub/pull/156.

**Pull request closed without merging**